docs/README.md

```diff
@@ -123,7 +123,7 @@ Example of all these handlers:
 ```js
 {
   $: '/foo/',
-  beforeEnter: ({oldPath, newPath, params}) => void | Promise<{ redirect: string, replace?: boolean }>,
```

Comment:
Intentionally not mentioned in the docs. The need to support `undefined` is essentially a type system noise that is better documented as just `void` ;)